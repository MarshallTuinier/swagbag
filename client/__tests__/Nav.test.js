import { mount } from "enzyme";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import toJSON from "enzyme-to-json";
import Nav from "../components/Nav";
import { CURRENT_USER_QUERY } from "../GraphQL";
import { fakeUser } from "../lib/testUtils";

const notSignedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { currentUser: null } }
  }
];

const signedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { currentUser: fakeUser() } }
  }
];

describe("<Nav/>", () => {
  it("renders a minimal nav when signed out", async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find("ul[data-test='nav']");
    expect(nav.children().length).toBe(2);
  });

  it("renders full nav when signed in", async () => {
    const wrapper = mount(
      <MockedProvider mocks={signedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find("ul[data-test='nav']");
    expect(nav.children().length).toBe(6);
    expect(nav.text()).toContain("Sign Out");
    const count = nav.find("div.count");
    expect(toJSON(count)).toMatchSnapshot();
  });
});
