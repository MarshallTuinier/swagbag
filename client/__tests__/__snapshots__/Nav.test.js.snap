// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> renders full nav when signed in 1`] = `
<div
  className="count CartCount__Dot-sc-1d760cr-0 fOUfXM"
>
  0
</div>
`;
